### PR TITLE
feat(user credentials): register thirt party platform and credentials

### PR DESCRIPTION
**Pull request by nahuelRo**

### CODE REVIEW

File: api/models/base.models.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const BaseSchema = new Schema<BaseEntity>(
   {
     tenantId: {
       type: Schema.Types.ObjectId,
-      required: true,
+      default: null,
```
Comment (sebikap-gm2) [inline api/models/base.models.ts:13]: Review:
Can we leave this as required, and manually specify which schema can support a null tenantId (such as UserSchema when User is first created)?

File: api/routes/auth.routes.ts [hunk 2]
```diff
@@ -2,8 +2,7 @@ import express from "express";
 export const authRoutes = express.Router();
 
 import { AuthController } from "../controllers";
-import { AuthMiddleware } from "../middleware/auth.middleware";
 
 authRoutes.post("/register", AuthController.register);
 authRoutes.post("/login", AuthController.login);
-authRoutes.get("/me", AuthMiddleware.validateUser, AuthController.me);
+authRoutes.post("/registerplatform", AuthController.registerPlatform);
```
Comment (sebikap-gm2) [inline api/routes/auth.routes.ts]: registerAuthenticationProvider

File: api/services/auth.services.ts [hunk 3]
```diff
@@ -2,17 +2,27 @@ import { UserRepository } from "../models";
 import { CreationUser, User } from "../types";
 
 export class AuthServices {
-  static async getUserbyEmail(email: User["email"]) {
-    const user = await UserRepository.findOne({ email: email }).exec();
+  static async validateIfExistEmail(email: User["email"]): Promise<void> {
```
Comment (sebikap-gm2) [inline api/services/auth.services.ts]: Don't specify, let TS infer

File: api/controllers/auth.controller.ts [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-import { TypedRequestBody, UserPayload } from "../types";
 import { NextFunction, Request, Response } from "express";
+import { UserRepository } from "../models";
 
 export class AuthController {
   static async register(req: Request, res: Response, next: NextFunction) {
     try {
       const { email } = req.body;
-      await AuthServices.getUserbyEmail(email);
+      await AuthServices.validateIfExistEmail(email);
```
Comment (sebikap-gm2) [inline api/controllers/auth.controller.ts]: AuthController should handle "email already exists" logic

File: api/types/user.ts [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
-type UserJWT = Pick<User, "_id" | "email">;
+export type RegisterUser = Pick<User, "name" | "email" | "password">;
+
+type UserJWT = Pick<UserEntity, "_id" | "email">;
 
 export type UserPayload = JwtPayload & UserJWT;
 
 export type CreationUser = Omit<User, "_id" | "__v">;
+
+export type UserEntity = Document & User & BaseEntity;
```
Comment (sebikap-gm2) [inline api/types/user.ts]: Move on top of UserJWT

Comment (sebikap-gm2) [reply api/types/user.ts]: BaseEntity already extends Document

File: api/utils/bcryptFunctions.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import bcrypt from "bcrypt";
+import { UserEntity } from "../types";
+
+export async function validatePassword(
+  user: UserEntity,
+  password: string
+): Promise<boolean> {
```
Comment (sebikap-gm2) [inline api/utils/bcryptFunctions.ts]: Let TS infer

File: api/utils/bcryptFunctions.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  const hashedPassword = await bcrypt.hash(password, user.salt);
+
+  if (user.password !== hashedPassword) {
+    throw new Error(`Wrong password! Please try again`);
```
Comment (sebikap-gm2) [inline api/utils/bcryptFunctions.ts]: Non specific password error message

**Review by sebikap-gm2 — CHANGES_REQUESTED** [review]

File: api/models/base.models.ts (hunk 1 above)
Comment (nahuelRo) [reply api/models/base.models.ts:13]: I was looking to solve this problem with moongose ​​and I didn't find much information, what I did in the user shema is not to inherit from baseEntity but to have the same types as the other schemas that do inherit from baseEntity

**Review by sebikap-gm2 — APPROVED** [review]

**Pull request merged**